### PR TITLE
Fix recipe test problems

### PR DESCRIPTION
**Pull request by freiksenet**
## Description

* Strip ansi colors on diff during tests
* Use tmpdir to run fixtures
* Fix snapshots having wrong fixture formatting

### CODE REVIEW

File: packages/gatsby-recipes/src/providers/gatsby/plugin.test.js [hunk 1]
```diff
@@ -1,24 +1,49 @@
...
+)
+const HELLO_WORLD_FIXTURE = path.join(
   __dirname,
   `./fixtures/gatsby-starter-hello-world`
 )
 const name = `gatsby-plugin-foo`
-const configPath = path.join(root, `gatsby-config.js`)
 
 describe(`gatsby-plugin resource`, () => {
+  let tmpDir
```
Comment (KyleAMathews) [inline packages/gatsby-recipes/src/providers/gatsby/plugin.test.js:20]: What problems was this causing?

File: packages/gatsby-recipes/src/providers/gatsby/__snapshots__/plugin.test.js.snap [hunk 2]
```diff
@@ -18,7 +18,7 @@ Object {
 module.exports = {
   /* Your site config here */
   plugins: [],
-}
+};
```
Comment (pvdz) [inline packages/gatsby-recipes/src/providers/gatsby/__snapshots__/plugin.test.js.snap:21]: What's up with the semi's in this file?

File: yarn.lock [hunk 3]
```diff
@@ -22685,11 +22685,6 @@ sudo-prompt@^8.2.0:
   resolved "https://registry.yarnpkg.com/sudo-prompt/-/sudo-prompt-8.2.5.tgz#cc5ef3769a134bb94b24a631cc09628d4d53603e"
   integrity sha512-rlBo3HU/1zAJUrkY6jNxDOC9eVYliG6nS4JA8u8KAshITd07tafMc/Br7xQwCSseXwJ2iCcHCE8SNWX3q8Z+kw==
 
-sudo-prompt@^8.2.0:
```
Comment (pvdz) [inline yarn.lock:22688]: :thinking:

File: packages/gatsby-recipes/src/providers/gatsby/plugin.test.js (hunk 1 above)
Comment (freiksenet) [reply packages/gatsby-recipes/src/providers/gatsby/plugin.test.js:20]: Test run modified fixtures without restoring them.

File: packages/gatsby-recipes/src/providers/utils/get-diff.js [hunk 4]
```diff
@@ -12,7 +13,11 @@ module.exports = async (oldVal, newVal) => {
     expand: false,
   }
 
-  const diffText = diff(oldVal, newVal, options)
+  let diffText = diff(oldVal, newVal, options)
+
+  if (process.env.GATSBY_RECIPES_NO_COLOR) {
```
Comment (wardpeet) [inline packages/gatsby-recipes/src/providers/utils/get-diff.js:18]: i'm wondering if we could do this in the test instead, and not in code? Or do we want people to have this option too?

File: packages/gatsby-recipes/src/providers/gatsby/plugin.test.js (hunk 1 above)
Comment (johno) [reply packages/gatsby-recipes/src/providers/gatsby/plugin.test.js:20]: Yeah, I think this is the result of prettier resolving a different config (or none at all) so writing back out the config after the last destroy results in a diff.

File: packages/gatsby-recipes/src/providers/gatsby/__snapshots__/plugin.test.js.snap (hunk 2 above)
Comment (KyleAMathews) [reply packages/gatsby-recipes/src/providers/gatsby/__snapshots__/plugin.test.js.snap:21]: probably cause we run prettier on the gatsby-config.js source after we add a plugin — so without a .prettierrc around for tests, it adds semis by default

File: packages/gatsby-recipes/src/providers/utils/get-diff.js (hunk 4 above)
Comment (KyleAMathews) [reply packages/gatsby-recipes/src/providers/utils/get-diff.js:18]: like pass it in as an option into the function? That wouldn't keep it any more hidden.

**Comment by KyleAMathews** [comment]
Doesn't seem like there's any objections to the changes & we need to get more recipes changes merged in so getting this in.

Thanks @freiksenet and others for figuring this out!

**Comment by freiksenet** [comment]
Lol, didn't realize we didn't merge it before.

**Pull request merged**